marlowe-cli/src/Language/Marlowe/CLI/Run.hs

```diff
@@ -384,16 +384,34 @@ prepareTransactionImpl marloweIn txInputs minimumTime maximumTime printStats =
 
 -- | Prepare the next step in a Marlowe contract.
 makeMarlowe :: MonadError CliError m
+            => MonadIO m
             => MarloweTransaction lang era                            -- ^ The Marlowe initial state and initial contract.
             -> TransactionInput                                       -- ^ The transaction input.
             -> m ([TransactionWarning], MarloweTransaction lang era)  -- ^ Action to compute the next step in the contract.
 makeMarlowe marloweIn@MarloweTransaction{..} transactionInput =
   do
-    transactionInput'@TransactionInput{..} <-
-      merkleizeInputs marloweIn transactionInput
-        `catchError` (const $ pure transactionInput)  -- TODO: Consider not catching errors here.
+    let
+      toSlot = posixTimeToEnclosingSlot mtSlotConfig
+      toSlotNo = SlotNo . fromIntegral . toSlot
+
+      roundTxInterval ti@TransactionInput{txInterval=txInterval@(minimumTime, maximumTime)} = do
+        let
+          txInterval' =
+            ( slotToBeginPOSIXTime mtSlotConfig . toSlot $ minimumTime
```

Comment:
This may work in most cases, but I'm skeptical that using `slotToBeginPOSIXTime` behaves exactly like the function that a nodes uses to convert them before passing them to the Plutus interpreter (`Cardano.Ledger.Alonzo.TxInfo.transVITime` in `cardano-ledger-alonzo`).

Reply:
So started with `transVITime` but brings here a lot of complexity which requires communication with `cardano-node`. Should I make another pass over this?

Reply:
I mean it uses a lot of pieces which we use in `querySlotConfig` plus protocol parameters (only to grab the `ProtocolVersion` under the hood) so it is of course feasible ;-)